src/alda/sound.clj

```diff
@@ -278,6 +288,61 @@
                                (/ (score-length events) 1000.0)
                                1) end!)))
 
+(defn schedule-wait!
```

Comment:
This is just an aesthetic nitpick, but I noticed that `schedule-wait!` has some repeated code from `schedule-events!`. I think it would be cleaner to incorporate this new way of scheduling events into `schedule-events!` directly. You could use the `*use-midi-sequencer*` dynamic var as a way of deciding whether to do the existing thing (the `pdoseq-block` form where we pipe events into the JSyn SynthesisEngine scheduler) or do this new thing (piping events into a Java MIDI Sequencer).

The part where we schedule the `end!` event could be used for both implementations -- the only thing that differs is the way we calculate the length of the score. So that might look something like:

```clojure
(let [...
      end!         #(deliver wait :done)
      score-length (if *use-midi-sequencer*
                     (/ (.getMicrosecondLength sequencer) 1000.0)
                     (score-length events))]
  (schedule-event! engine (+ begin (/ score-length 1000.0) 1) end!))
```

Reply:
Sure -- we could store it in a top-level var `*midi-sequencer*` that is `nil` by default.

Now that I think about it, that could actually be what we use (instead of having another, separate value, `*use-midi-sequencer*`) to determine whether or not to use the MIDI sequencer:

```clojure
(if *midi-sequencer*
  (/ (.getMicrosecondLength *midi-sequencer*) 1000.0)
  (score-length events))
```

Reply:
We could store it in alda.sound.midi, the same place where we keep `*midi-synth*`: https://github.com/alda-lang/alda-sound-engine-clj/blob/master/src/alda/sound/midi.clj#L20

Reply:
I tried doing this, but it turned out that trying to play back multiple sequences seems to have interesting effects, it seems like the volume was stacking up on itself for some reason. I also noticed that creating sequencers seem to have no pause (since they don't have to playback resources), so I think I'll just create/dispose of sequencers as I need to. Sharing the synth with the main instance seems to be good enough, from what I've seen.

Reply:
OK -- that sounds reasonable to me.